packages/ai/src/ai/web/metrics/tracing.py

```
@@ -0,0 +1,116 @@
+def setup_tracing(app=None) -> TracerProvider:
+    """Initialise the OpenTelemetry TracerProvider and optionally instrument a FastAPI app.
+
+    Args:
+        app: A FastAPI application instance.  When provided the
+            ``opentelemetry-instrumentation-fastapi`` instrumentor will
+            automatically create spans for every inbound request.
+
+    Returns:
+        The configured ``TracerProvider``.
+    """
+    global _tracer_provider
```

Ruff (0.15.7)

[warning] Using the global statement to update `_tracer_provider` is discouraged

(PLW0603)